src/concrete/ml/quantization/post_training.py

```diff
@@ -23,6 +23,97 @@
 from .quantized_ops import QuantizedBrevitasQuant
 from .quantizers import QuantizationOptions, QuantizedArray, UniformQuantizer
 
+# pylint: disable=too-many-lines
+
+
+def _inspect_tree_n_bits(n_bits):
+    """Validate the 'n_bits' parameter for tree-based models.
+
+    This function checks whether 'n_bits' is a valid integer or dictionary.
+    - If 'n_bits' is an integer, it must be a non-null positive, its value is assigned to
+        "op_inputs" and "op_leaves" bits
+    - If it is a dictionary, it should contain integer values for keys 'op_leaves' and 'op_inputs',
+        where 'op_leaves' should not exceed 'op_inputs'.
+
+    The function raises a ValueError with a descriptive message if 'n_bits' does not meet
+    these criteria.
+
+    Args:
+        n_bits (int, Dict[str, int]): number of bits for quantization, can be a single value or
+            a dictionary with the following keys :
+            - "op_inputs" (mandatory): number of bits to quantize the input values
+            - "op_leaves" (optional): number of bits to quantize the leaves, must be less than or
+                equal to `op_inputs`. defaults to the value of "op_inputs" if not specified.
+
+    Raises:
+        ValueError: If 'n_bits' does not conform to the required format or value constraints.
+    """
+
+    detailed_message = (
+        "Invalid `n_bits`, either pass a non-null positive integer or a dictionary containing "
```

Comment:
back quotes / backicks (`) have a special meaning in python when printing them and it's supposed to be deprecated in python 3.0: https://stackoverflow.com/questions/1673071/what-do-backticks-mean-to-the-python-interpreter-example-num

in our case I think we just want to use single quotes (') instead here as we only want to refer to the name ! this goes for all or our error messages ect I've created an issue for it as I think we already have some other cases in the code : https://github.com/zama-ai/concrete-ml-internal/issues/4235

Reply:
ok thanks